test/integration/dai.spec.js

```diff
@@ -135,50 +136,96 @@ test(
   300000
 );
 
-test(
-  'can sell Dai',
-  async () => {
-    let tx, error;
-
-    if (process.env.NETWORK === 'test') {
-      await createDaiAndPlaceLimitOrder(exchange);
-    }
-
-    try {
-      tx = await exchange.sellDai('0.1', WETH);
-    } catch (err) {
-      console.error(err);
-      error = err;
-    }
-
-    expect(tx).toBeDefined();
-    expect(error).toBeUndefined();
-  },
-  600000
-);
+xdescribe('OasisExchangeService', () => {
+  test(
+    'can sell Dai',
+    async () => {
+      let tx, error;
+
+      if (process.env.NETWORK === 'test') {
+        await createDaiAndPlaceLimitOrder(exchange);
+      }
+
+      try {
+        tx = await exchange.sellDai('0.1', WETH);
+      } catch (err) {
+        console.error(err);
+        error = err;
+      }
+
+      expect(tx).toBeDefined();
+      expect(error).toBeUndefined();
+    },
+    600000
+  );
+
+  test(
+    'can buy Dai',
+    async () => {
+      let tx, error;
+      await checkWethBalance();
+
+      if (process.env.NETWORK === 'test') {
+        await createDaiAndPlaceLimitOrder(exchange, true);
+      }
+
+      try {
+        tx = await exchange.buyDai('0.1', WETH);
+      } catch (err) {
+        console.error(err);
+        error = err;
+      }
+
+      expect(tx).toBeDefined();
+      expect(error).toBeUndefined();
+    },
+    600000
+  );
+});
 
-test(
-  'can buy Dai',
-  async () => {
-    let tx, error;
-    await checkWethBalance();
-
-    if (process.env.NETWORK === 'test') {
-      await createDaiAndPlaceLimitOrder(exchange, true);
-    }
-
-    try {
-      tx = await exchange.buyDai('0.1', WETH);
-    } catch (err) {
-      console.error(err);
-      error = err;
-    }
-
-    expect(tx).toBeDefined();
-    expect(error).toBeUndefined();
-  },
-  600000
-);
+// These will fail because of some issues with the transaction
+// object's logging functions, maybe because of dsproxy
+xdescribe('Eth2DaiDirect', () => {
+  test(
+    'sell eth for dai',
+    async () => {
+      const order = await maker.service('exchange').sell('ETH', 'DAI', '0.01');
```

Comment:
@levity This and the other four functions in this describe block are where we sometimes hit the txObject errors. Any of these transactions seems to randomly have one of three outcomes: the txObject works as expected, the txObject throws a TypeError `Cannot read property 'gasUsed' of null` (where `this.receipt` is `null`), and `Tried getTransaction 10 times and still failed` (these errors seem very related). All of these transactions actually work (verified on etherscan).

I confirmed with Michael that neither of us have checked the library's proxy functionality on mainnet yet, so I'm planning to add some of the proxy cdp tests to this file to see if we get the same behavior.

There's also some logging info built into `OasisOrder` that doesn't get all the data it should (e.g. `order.fillAmount()` is always 0, but `order.fees()` works), so that could also be related.

Reply:
Also worth noting that I've only seen these errors in this context on mainnet

Reply:
> All of these transactions actually work (verified on etherscan)

I see... I think there may have been a regression when I added some of Patrick's changes from the websockets branch. Try applying https://github.com/makerdao/dai.js/pull/78 to your branch and see if at least the "Tried getTransaction 10 times and still failed" errors go away.

Reply:
By the way, this did work for the "tried 10 times" error, but I'm still randomly not getting a value for this.receipt